crates/primitives/starknet/src/sequencer_address/mod.rs

```diff
@@ -0,0 +1,100 @@
+use scale_codec::{Decode, Encode};
+use sp_inherents::{InherentData, InherentIdentifier, IsFatalError};
+
+/// The identifier for the `sequencer_address` inherent.
+pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"seqaddr0";
+
+/// Default value in case the sequencer address is not set.
+pub const DEFAULT_SEQUENCER_ADDRESS: [u8; 32] =
+    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 222, 173];
+
+/// The inherent type for the sequencer address.
+pub type InherentType = [u8; 32];
+
+#[derive(Decode, Encode, sp_runtime::RuntimeDebug)]
+/// Error types when working with the sequencer address.
+pub enum InherentError {
+    /// Submitted address must be `[u8; 32]`.
+    WrongAddressFormat,
+}
+
+impl IsFatalError for InherentError {
+    fn is_fatal_error(&self) -> bool {
+        match self {
+            InherentError::WrongAddressFormat => true,
+        }
+    }
+}
+
+impl InherentError {
+    #[cfg(feature = "std")]
```

Comment:
Why only `std`?

It really required you can put it above the whole `impl InherentError` block